File: Calculator/ios-native/CalculatorHub/Calculators/DiscountCalculator.swift
import Foundation

struct DiscountResult {
    let savedAmount: Double
    let finalPrice: Double
}

class DiscountCalculator {
    func calculate(originalPrice: Double, discountPercentage: Double) -> DiscountResult {
        let saved = originalPrice * (discountPercentage / 100.0)
        let finalPrice = originalPrice - saved
        
        return DiscountResult(
            savedAmount: saved,
            finalPrice: finalPrice
        )
    }
}

File: Calculator/android-native/app/src/main/java/com/qaantum/calculatorhub/calculators/DiscountCalculator.kt
package com.qaantum.calculatorhub.calculators

data class DiscountResult(
    val savedAmount: Double,
    val finalPrice: Double
)

class DiscountCalculator {
    fun calculate(
        originalPrice: Double,
        discountPercentage: Double
    ): DiscountResult {
        val saved = originalPrice * (discountPercentage / 100.0)
        val finalPrice = originalPrice - saved
        
        return DiscountResult(
            savedAmount = saved,
            finalPrice = finalPrice
        )
    }
}
